### PR TITLE
allow splash image to be loaded from SD on Horus

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
This basically implements issue #5434 as described. The image `/IMAGES/splash.png` is loaded from the SD card while booting and used as a splash image. If the image is not present, nothing changes and the splash image embedded in the firmware is used.

Things tested:
- no file: nothing happens (default is loaded: it's optional after all).
- wrong file (some text inside): nothing happens (default is loaded).
- big file (640x480): picture is probably cropped, because I see the background.

While testing, I did not notice any usual delay compared to the normal splash image.

The splash screen can be modified by using the new menu item "Splash screen" in the SD card manager. That will copy the file to `/IMAGES/splash.png`, thus overwriting any existing file with the same name.

### CODE REVIEW

**Comment by Barbaryann** [comment]
It is only working on the next 2.2.2 version ?
On the 2.2.1 we can't change the slpash screen on our X10 ?

**Comment by raphaelcoeffic** [comment]
@Barbaryann exactly. If you want it earlier, you'll have to merge/compile yourself.

**Comment by Barbaryann** [comment]
Thanks but i don't really know how to do that, I can wait some week to have a nice Splash screen.
Thanks

File: radio/src/gui/480x272/splash.cpp [hunk 1]
```diff
@@ -19,13 +19,36 @@
...
+
 void drawSplash()
 {
-  lcd->clear();
-  lcd->drawBitmap((LCD_W-BMP_SPLASH.getWidth())/2, (LCD_H-BMP_SPLASH.getHeight())/2, &BMP_SPLASH);
+    static bool     loadImgFromSD  = true;
+    static BitmapBuffer* splashImg = NULL;
+
+    if (loadImgFromSD && splashImg == NULL) {
+        bool sd_mounted = sdMounted();
```
Comment (bsongis) [inline radio/src/gui/480x272/splash.cpp:33]: is it possible to go there? if yes would you add a comment?

Comment (raphaelcoeffic) [reply radio/src/gui/480x272/splash.cpp:33]: It definitely is, that's what happens the very first time the splash function is called. The goal is to execute this code only once, whether it failed or not, we should not re-try later.

Comment (raphaelcoeffic) [reply radio/src/gui/480x272/splash.cpp:33]: But normally, the SD card should be unmounted when splash() is called the first time. So basically the case where the SD is already mounted is only handled here as a safeguard.

**Comment by trojanGoat** [comment]
Hey Guys, 

This is only cosmetic improvements, but it would be very cool to be able to use a .gif file for the splash screen.  

Actually if you guys are interested, I can volunteer my time and make a 3d animated/rendered splash screen for open tx.  Here's a little example of what I made for our group.  https://www.youtube.com/watch?v=s06MtPKOwKc

**Comment by sly1231** [comment]
_The splash screen can be modified by using the new menu item "Splash screen" in the SD card manager. That will copy the file to /IMAGES/splash.png, thus overwriting any existing file with the same name._


Where does one find ""Splash screen" in the SD card manager"

**Comment by kilrah** [comment]
In yet unreleased version 2.2.2

**Pull request merged**